src/main/java/com/synopsys/integration/detect/lifecycle/run/step/ContainerScanStepRunner.java

```diff
@@ -4,25 +4,92 @@
 import java.io.IOException;
 import java.util.UUID;
 
-import com.synopsys.integration.detect.configuration.DetectUserFriendlyException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 import com.synopsys.integration.detect.lifecycle.run.data.BlackDuckRunData;
 import com.synopsys.integration.detect.lifecycle.run.operation.OperationRunner;
-import com.synopsys.integration.detect.util.bdio.protobuf.DetectProtobufBdioUtil;
+import com.synopsys.integration.detect.util.bdio.protobuf.DetectProtobufBdioHeaderUtil;
 import com.synopsys.integration.exception.IntegrationException;
+import com.synopsys.integration.rest.response.Response;
+import com.synopsys.integration.util.NameVersion;
+
+import net.minidev.json.JSONObject;
 
 public class ContainerScanStepRunner {
 
     private final OperationRunner operationRunner;
     private UUID scanId;
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+    private final NameVersion projectNameVersion;
+    private final BlackDuckRunData blackDuckRunData;
+    private final File binaryRunDirectory;
 
-    public ContainerScanStepRunner(OperationRunner operationRunner) throws DetectUserFriendlyException {
+    public ContainerScanStepRunner(OperationRunner operationRunner, NameVersion projectNameVersion, BlackDuckRunData blackDuckRunData) {
         this.operationRunner = operationRunner;
+        this.projectNameVersion = projectNameVersion;
+        this.blackDuckRunData = blackDuckRunData;
+        binaryRunDirectory = operationRunner.getDirectoryManager().getBinaryOutputDirectory();
     }
 
-    public UUID initiateScan(BlackDuckRunData blackDuckRunData) throws IOException, IntegrationException {
-        DetectProtobufBdioUtil detectProtobufBdioUtil = new DetectProtobufBdioUtil(UUID.randomUUID().toString(), "CONTAINER");
-        File bdioHeaderFile = detectProtobufBdioUtil.createProtobufBdioHeader();
-        scanId = operationRunner.uploadBdioHeaderToInitiateScan(blackDuckRunData, bdioHeaderFile);
+    public UUID invokeContainerScanningWorkflow() throws IntegrationException, IOException {
+        initiateScan();
+        uploadImageToStorageService();
+        uploadImageMetadataToStorageService();
         return scanId;
     }
+
+    public void initiateScan() throws IOException, IntegrationException {
+//        File bdioHeaderFile = new File(Application.class.getResource("/test-inputs/bdio-header.pb").getPath()); // temporary
+        DetectProtobufBdioHeaderUtil detectProtobufBdioHeaderUtil = new DetectProtobufBdioHeaderUtil(UUID.randomUUID().toString(), "CONTAINER", projectNameVersion);
+        File bdioHeaderFile = detectProtobufBdioHeaderUtil.createProtobufBdioHeader(binaryRunDirectory);
+        scanId = operationRunner.uploadBdioHeaderToInitiateScan(blackDuckRunData, bdioHeaderFile);
+        String scanIdString = scanId.toString();
```

Comment:
Probably personal preference but I wouldn't do the assignment just to use it in the logging statement as it's pretty straightfoward.

Reply:
My own preference is not to, to avoid NPE in log unless NPE cannot happen.

Reply:
Yeah having any sort of evaluation in the logger is picked up by SonarCloud as a "Major" category code smell (as @gopannabd mentioned, maybe it picks it up to avoid NPEs in loggers). I too had `scanId.toString` directly in the logger statement initially :)